src/CMakeLists.txt

```diff
@@ -87,62 +86,78 @@ if(CCTAG_WITH_CUDA)
   # set(BUILD_SHARED_LIBS ON)
 
   set(CUDA_NVCC_FLAGS  "${CUDA_NVCC_FLAGS};-DCUB_CDP")
- 
+
   # this must go before CUDA_ADD_LIBRARY otherwise we won't be able to add it
   # after
   # https://github.com/Kitware/CMake/blob/master/Modules/FindCUDA.cmake#L147
-  # INCLUDE_DIRECTORIES -- "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}>" 
+  # INCLUDE_DIRECTORIES -- "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}>"
   include_directories(${CMAKE_CURRENT_LIST_DIR}
-                      ${Boost_INCLUDE_DIRS} ${OpenCV_INCLUDE_DIRS} ${Eigen_INCLUDE_DIR}
+                      ${OpenCV_INCLUDE_DIRS} ${Eigen_INCLUDE_DIR}
 		      ${TBB_INCLUDE_DIRS}
                       "${CMAKE_SOURCE_DIR}/src"
                       "${CMAKE_SOURCE_DIR}/3rdparty/Cuda-7.0-cub")
- 
+
   CUDA_ADD_LIBRARY(CCTag ${CUDA_cpp} ${CUDA_cu} ${CCTag_cpp})
 
   # since we used CUDA_ADD_LIBRARY we cannot use PUBLIC or PRIVATE here
-  target_link_libraries(CCTag 
-                       ${OpenCV_LIBS} ${Boost_LIBRARIES} ${Eigen_LIBRARIES} 
-                       ${TBB_LIBRARIES} ${CUDA_CUDADEVRT_LIBRARY} pthread dl)
+  target_link_libraries(CCTag
+                       ${OpenCV_LIBS}
+                       Boost::date_time Boost::chrono Boost::thread Boost::serialization Boost::system Boost::filesystem Boost::atomic Boost::program_options Boost::timer
+                       ${Eigen_LIBRARIES}
+                       ${TBB_LIBRARIES} ${CUDA_CUDADEVRT_LIBRARY})
+
+  if(NOT "${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
+   target_link_libraries(CCTag pthread dl)
+  endif()
 
-  target_compile_definitions(CCTag 
-                         PUBLIC ${Boost_DEFINITIONS} 
+  target_compile_definitions(CCTag
                          PUBLIC -DCCTAG_WITH_CUDA -DCUB_CDP
-                         PRIVATE ${TBB_DEFINITIONS}) 
+                         PRIVATE ${TBB_DEFINITIONS})
 
   # This is nececessary for the CCTagConfig.cmake to correctly export the
   # includes, always because we used CUDA_ADD_LIBRARY
-  set_target_properties(CCTag PROPERTIES INTERFACE_INCLUDE_DIRECTORIES 
+  set_target_properties(CCTag PROPERTIES INTERFACE_INCLUDE_DIRECTORIES
        "$<INSTALL_INTERFACE:$<INSTALL_PREFIX>/include>;${CUDA_INCLUDE_DIRS};${Boost_INCLUDE_DIRS};${Eigen_INCLUDE_DIR};${OpenCV_INCLUDE_DIRS}")
   get_target_property(testprop CCTag INTERFACE_INCLUDE_DIRECTORIES )
   message(STATUS "testprop: ${testprop}")
 
-else(CCTAG_WITH_CUDA)
+else() # without CUDA
 
   set(CUDA_cpp "")
 
   message( STATUS "Declare CCTag library" )
   add_library(CCTag ${CCTag_cpp})
   set_target_properties(CCTag PROPERTIES VERSION ${PROJECT_VERSION})
 
-  target_include_directories(CCTag 
+  target_include_directories(CCTag
                            PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}>"
                                   "$<INSTALL_INTERFACE:$<INSTALL_PREFIX>/include>"
                            PUBLIC ${Boost_INCLUDE_DIRS} ${Eigen_INCLUDE_DIR} ${OpenCV_INCLUDE_DIRS}
```

Comment:
${Boost_INCLUDE_DIRS} can now be removed. When you specify the target Boost::boost in `target_link_libraries()` it automagically add the relevant includes and definitions